test/cppsuite/test_harness/workload/database_operation.h

```diff
@@ -97,95 +85,133 @@ class database_operation {
          * here.
          */
         for (int64_t i = 0; i < thread_count; ++i) {
-            int64_t collections_per_thread = collection_count / thread_count;
-            std::vector<std::string> thread_collections;
-            for (size_t j = i * collections_per_thread;
-                 j < i * collections_per_thread + collections_per_thread; j++) {
-                debug_print("Populate: adding collection: " + collection_names[j] + " to thread " +
-                    std::to_string(i),
-                  DEBUG_TRACE);
-                thread_collections.push_back(collection_names[j]);
-            }
-            tm.add_thread(populate_worker, i, thread_collections,
-              connection_manager::instance().create_session(), tsm, tracking, key_count, key_size,
-              value_size);
+            thread_context *tc =
```

Comment:
We should refactor at some point and use `std::unique_ptr` for this type of thing.